docs/chains/deps-and-resources.mdx

```diff
@@ -0,0 +1,192 @@
+
+Before we introduce the Mistral code, let's first define the resources required for this chain.
+
+```python
+
+import truss_chains as chains
+import pydantic
+from truss import truss_config
+
+
+class MistraLLMConfig(pydantic.BaseModel):
+    hf_model_name: str
+
+class MistralLLM(chains.ChainletBase[MistraLLMConfig]):
+    # The RemoteConfig object defines the resources required for this chainlet
+    remote_config = chains.RemoteConfig(
+        # The DockerImage object defines properties of the docker image.
+        # Here, we can define the pip and system dependencies required for the chainlet.
+        docker_image=chains.DockerImage(
+            pip_requirements=[
+                "git+https://github.com/basetenlabs/truss.git@9be68f563d735289128cd3c8853bdf62ec03cef3"
+                "transformers==4.38.1",
+                "torch==2.0.1",
+                "sentencepiece",
+                "accelerate",
+            ],
+        ),
+        # The Compute object defines the compute resources required for the chainlet.
+        # Here, we specify that we require an A10G GPU
+        compute=chains.Compute(cpu_count=2, gpu="A10G"),
+    )
+
+    def __init__(
+        self,
+        context: chains.DeploymentContext = chains.provide_context(),
+    ) -> None:
+        super().__init__(context)
+        ...
+
+
+    def run(self, data: str) -> str:
+        ...
+
+```
+
+# The Mistral Code
+
+Once we've defined the needed resources, we can now add the Mistral code to the chainlet.
+
+```python
+import truss_chains as chains
+from truss import truss_config
+import pydantic
+
+MISTRAL_HF_MODEL = "mistralai/Mistral-7B-Instruct-v0.2"
+
+class MistraLLMConfig(pydantic.BaseModel):
+    hf_model_name: str
+
+class MistralLLM(chains.ChainletBase[MistraLLMConfig]):
+    remote_config = chains.RemoteConfig(
+        docker_image=chains.DockerImage(
+            pip_requirements=[
+                "git+https://github.com/basetenlabs/truss.git@9be68f563d735289128cd3c8853bdf62ec03cef3",
+                "transformers==4.38.1",
+                "torch==2.0.1",
+                "sentencepiece",
+                "accelerate",
+            ],
+        ),
+        compute=chains.Compute(cpu_count=2, gpu="A10G"),
+        # This is one of the main things that we've added. This shows off using
+        # the Model Cache feature of Truss to cache the weights of the model at
+        # build time.
+        assets=chains.Assets(cached=[
+            truss_config.ModelRepo(
+                repo_id=MISTRAL_HF_MODEL, allow_patterns=["*.json", "*.safetensors", ".model"]
+            )
+        ]),
+    )
+    default_user_config = MistraLLMConfig(hf_model_name=MISTRAL_HF_MODEL)
+
+    def __init__(
+        self,
+        context: chains.DeploymentContext[MistraLLMConfig] = chains.provide_context(),
+    ) -> None:
+        super().__init__(context)
+        # Note that the code in the __init__ and run methods are
```

Comment:
This comment is not clear.

Reply:
Instead, we need to call out that transformers and torch are not imported on top-level (so that `PoemGenerator` won't crash in it's remote env without these deps).